pkg/policies/action/action.go

```diff
@@ -0,0 +1,757 @@
+	// evaluated before deny rules at each priority tier.
+	Rules []*Rule `json:"rules"`
+}
+
+// Rule is an Action Use rule
+type Rule struct {
+	// Name is the name used to identify the rule
+	Name string `json:"name"`
+
+	// Method is the type of rule. One of "require", "allow", and "deny".
+	Method string `json:"method"`
+
+	// Priority is the priority tier identifier applied to the rule.
+	// Options are "urgent", "high", "medium", and "low"
+	Priority string `json:"priority"`
+
+	// Actions is a set of ActionSelectors.
+	// If nil, all Actions will be selected
+	Actions []*ActionSelector `json:"actions"`
+
+	// MustPass specifies whether the rule's Action(s) are required to
+	// be part of a passing workflow on latest commit
+	MustPass bool `json:"mustPass"`
+
+	// RequireAll specifies that all Actions listed should be required,
+	// rather than just one.
+	// [For use with "require" method]
+	RequireAll bool `json:"requireAll"`
+
+	// group references the RuleGroup to which this rule belongs
+	group *RuleGroup
+
+	// priorityInt is an int corresponding to Priority.
+	// Lower number = higher priority
+	priorityInt int
+}
+
+// RepoSelector specifies a selection of repos
+type RepoSelector struct {
+	// Name is the repo name in glob format
+	Name string `json:"name"`
+
+	// Language is a set of programming languages.
+	// See the section about language detection below
+	Language []string `json:"language"`
+
+	// Exclude is a set of RepoSelectors targeting repos that should
+	// not be matched by this selector.
+	Exclude []*RepoSelector `json:"exclude"`
+}
+
+// ActionSelector specifies a selection of Actions
+type ActionSelector struct {
+	// Name is the Action name in glob format
+	Name string `json:"name"`
+
+	// Version is a semver condition or commit ref
+	// Default "" targets any version
+	Version string `json:"version"`
+}
+
+type details struct {
+	FailedRules []*Rule
+}
+
+type workflowMetadata struct {
+	filename string
+	workflow *actionlint.Workflow
+}
+
+type actionMetadata struct {
+	name             string
+	version          string
+	workflowFilename string
+	workflowName     string
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, config.ConfigLevel, interface{}) error
+
+var listWorkflows func(ctx context.Context, c *github.Client, owner, repo string, on []string) ([]*workflowMetadata, error)
+var listLanguages func(ctx context.Context, c *github.Client, owner, repo string) (map[string]int, error)
+var listWorkflowRunsByFilename func(ctx context.Context, c *github.Client, owner, repo string, workflowFilename string) ([]*github.WorkflowRun, error)
+var getLatestCommitHash func(ctx context.Context, c *github.Client, owner, repo string) (string, error)
+var listReleases func(ctx context.Context, c *github.Client, owner, repo string) ([]*github.RepositoryRelease, error)
+
+func init() {
+	configFetchConfig = config.FetchConfig
+	listWorkflows = githubListWorkflows
```

Comment:
The paradigm we've been using (not sure if I like it) is to have the real func be `listWorkflowsReal`